src/upload.rs

```diff
@@ -1,3 +1,6 @@
+//! This module handles the expected information a crate should have
+//! and manages the serialising and deserialising of this information
```

Comment:
Maybe we could look into moving the serialize stuff into the `tests/` module at some point?